common/src/state/pending_message.rs

```diff
@@ -0,0 +1,182 @@
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use once_cell::sync::Lazy;
+use tokio::sync::{
+    mpsc::{UnboundedReceiver, UnboundedSender},
+    Mutex,
+};
+use uuid::Uuid;
+use warp::{constellation::Progression, crypto::DID};
+
+use crate::warp_runner::ui_adapter::Message;
+
+pub struct AttachmentProgress {
+    pub progress: Progression,
+    pub conversation_id: Uuid,
+    pub msg: PendingSentMessage,
+}
+
+pub struct PendingMessageChannels {
+    pub tx: UnboundedSender<AttachmentProgress>,
+    pub rx: Arc<Mutex<UnboundedReceiver<AttachmentProgress>>>,
+}
+
+pub static MESSAGE_CHANNEL: Lazy<PendingMessageChannels> = Lazy::new(|| {
+    let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
+    PendingMessageChannels {
+        tx,
+        rx: Arc::new(Mutex::new(rx)),
+    }
+});
+
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct PendingSentMessages {
```

Comment:
it is a struct cause it has functions for it that do things instead of directly operating on the vec. can rewrite it to be not struct bound if you like but i just thought this way its more clearer for organization

Reply:
I appreciate that. I think that in this case, if we need to do that, the functions could be placed within `chats.rs`. But for now, I'd prefer removing the struct. 

It feels like `PendingSentMessages` simplified some things with `PendingSentMessage` which shouldn't have been there to begin with, and that we  won't need `PendingSentMessages`  if `PendingSentMessage` is removed. More about that - I think the `text` and `attachments` fields are redundant and that they are both part of `raygun::Message`. I also think it's incorrect to compare `PendingSentMessage` using the `text` instead of the message uuid (because two messages could have the same text). It seems that replacing `PendingSentMessage` with a new struct containing the message Uuid and a list of `warp::constellation::Progression` could be a lot simpler.

Reply:
the uuid problem is though that pending messages are actually dummy messages. the uuid of the actual message that will come from warp will be different than the dummy one. 
we can use the uuid in cases where we update attachment progress (as at that point we will have a ref to the dummy msg uuid) but not in cases where warp will send a "Message Sent Successfully" event to ui.

Reply:
when we send a message the list of attachment is passed as a string vec. warp then converts that into a vec of file (not uplink). also `File` here is a struct from warp and not the one from std lib in rust. it doesnt make much sense to also create dummy files when only thing relevant is the string here.